tests/optimizer/nonlinear/test_trust_region.py

```diff
@@ -0,0 +1,29 @@
+import pytest
+
+import theseus as th
+import torch
+
+
+@pytest.mark.parametrize("dof", [1, 8])
+def test_rho(dof):
+    # This test checks that rho = 1 for a simple linear problem min (x - t) ** 2
+    x = th.Vector(tensor=torch.zeros(1, dof), name="x")
+    t = th.Vector(tensor=torch.ones(1, dof), name="t")
+    o = th.Objective()
+    o.add(th.Difference(x, t, th.ScaleCostWeight(1.0), name="cf"))
```

Comment:
Can we add multiple factors with different `ScaleCostWeight`? In the future, we might create a factor to be something like `A*X-b`?